Fuzzer test fail in travis

https://travis-ci.org/Scille/parsec-cloud/builds/507434077

 ```
________________________________ test_fuzz_core ________________________________
[gw0] linux -- Python 3.6.3 /home/travis/virtualenv/python3.6.3/bin/python
value = <trio._core._run.Nursery object at 0x7f0718697b00>
    async def yield_(value=None):
>       return await _yield_(value)
../../../../virtualenv/python3.6.3/lib/python3.6/site-packages/async_generator/_impl.py:106: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../virtualenv/python3.6.3/lib/python3.6/site-packages/async_generator/_impl.py:99: in _yield_
    return (yield _wrap(value))
../tests/fuzzing/test_fuzz_core.py:243: in test_fuzz_core
    await trio.sleep(FUZZ_TIME)
../../../../virtualenv/python3.6.3/lib/python3.6/site-packages/trio/_core/_run.py:506: in __aexit__
    raise combined_error_from_nursery
../tests/fuzzing/test_fuzz_core.py:124: in fuzzer
    await _fuzzer_cmd(id, core, fs_state)
../tests/fuzzing/test_fuzz_core.py:154: in _fuzzer_cmd
    ret = await core.fs.file_read(path, size=size)
../parsec/core/fs/fs.py:103: in file_read
    return await self.file_fd_read(fd, size)
../parsec/core/fs/fs.py:128: in file_fd_read
    return await self._load_and_retry(self._local_file_fs.read, fd, size, offset)
../parsec/core/fs/fs.py:65: in _load_and_retry
    return fn(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <parsec.core.fs.local_file_fs.LocalFileFS object at 0x7f07186f8ba8>
fd = 8, size = 599, offset = None
    def read(self, fd: FileDescriptor, size: int = inf, offset: int = None) -> bytes:
        cursor = self._get_cursor_from_fd(fd)
    
        if offset is not None:
            self._seek(cursor, offset)
    
        hf = self._get_hot_file(cursor.access)
        if cursor.offset > hf.size:
            return b""
    
        manifest = self.local_folder_fs.get_manifest(cursor.access)
        assert is_file_manifest(manifest)
    
        start = cursor.offset
        end = cursor.offset + size
        if end > hf.size:
            end = hf.size
    
        blocks = self._get_quickly_filtered_blocks(manifest, start, end)
        blocks += hf.pending_writes
    
        merged = merge_buffers_with_limits(blocks, start, end)
        assert merged.size <= size
        assert merged.start == start
    
        missing = []
        data = bytearray(end - start)
        for cs in merged.spaces:
            for bs in cs.buffers:
                if isinstance(bs.buffer, DirtyBlockBuffer):
                    access = bs.buffer.access
                    try:
                        buff = self.get_block(access)
                    except LocalDBMissingEntry as exc:
>                       raise RuntimeError(f"Unknown local block `{access.id}`") from exc
E                       RuntimeError: Unknown local block `88d53381-c169-4596-9cdf-9efd55b0344a`
../parsec/core/fs/local_file_fs.py:260: RuntimeError
```

--- a/.codecov.yml
+++ b/.codecov.yml
@@ -1,0 +1,4 @@
+coverage:
+  status:
+    project: off
+    patch: off

--- a/parsec/core/fs/file_syncer.py
+++ b/parsec/core/fs/file_syncer.py
@@ -13,6 +13,7 @@ from parsec.core.types import (
     ManifestAccess,
     BlockAccess,
 )
+from parsec.core.fs.local_folder_fs import FSEntryNotFound
 from parsec.core.fs.merge_folders import find_conflicting_name_for_child_entry
 from parsec.core.fs.buffer_ordering import merge_buffers_with_limits_and_alignment
 from parsec.core.fs.local_file_fs import Buffer, DirtyBlockBuffer, BlockBuffer, NullFillerBuffer
@@ -64,6 +65,7 @@ class FileSyncerMixin(BaseSyncer):
             while buffers:
                 bs = buffers.pop()
                 if isinstance(bs.buffer, BlockBuffer):
+                    # TODO: look in local storage first
                     buff = await self._backend_block_read(bs.buffer.access)
                 elif isinstance(bs.buffer, DirtyBlockBuffer):
                     buff = self.local_file_fs.get_block(bs.buffer.access)
@@ -198,6 +200,14 @@ class FileSyncerMixin(BaseSyncer):
             blocks=blocks, size=sync_map.size  # TODO: useful ?
         )
 
+        # Make sure the corresponding entry still exists
+        # TODO: this is a hack, this should be replaced at some point
+        try:
+            self.local_folder_fs.get_entry_path(access.id)
+        except FSEntryNotFound:
+            logger.warning("Error while synchronizing a non-referenced file", access_id=access.id)
+            return
+
         # Upload the file manifest as new vlob version
         try:
             if is_placeholder_manifest(manifest):
@@ -232,6 +242,16 @@ class FileSyncerMixin(BaseSyncer):
             # await self._backend_vlob_create(access, to_sync_manifest, notify_beacons)
         else:
 
+            # Make sure the corresponding entry still exists
+            # TODO: this is a hack, this should be replaced at some point
+            try:
+                self.local_folder_fs.get_entry_path(access.id)
+            except FSEntryNotFound:
+                logger.warning(
+                    "Error while synchronizing a non-referenced file", access_id=access.id
+                )
+                return
+
             # The vlob has been successfully uploaded - clean up the dirty blocks
             for dirty_block in manifest.dirty_blocks:
                 self.local_file_fs.clear_dirty_block(dirty_block)
@@ -243,7 +263,7 @@ class FileSyncerMixin(BaseSyncer):
 
             self._sync_file_merge_back(access, manifest, to_sync_manifest)
 
-        return to_sync_manifest
+        return
 
     async def _sync_file(self, path: FsPath, access: Access, manifest: LocalFileManifest) -> None:
         """

--- a/parsec/core/fs/local_folder_fs.py
+++ b/parsec/core/fs/local_folder_fs.py
@@ -686,6 +686,7 @@ class LocalFolderFS:
         # corresponding accesses
 
         def _recursive_process_copy_map(copy_map):
+            access = copy_map["access"]
             manifest = copy_map["manifest"]
 
             cpy_access = ManifestAccess()
@@ -709,6 +710,16 @@ class LocalFolderFS:
                 cpy_manifest = LocalFolderManifest(author=self.local_author, children=cpy_children)
 
             self.set_dirty_manifest(cpy_access, cpy_manifest)
+
+            # TODO: If a file is opened while being moved, we cannot
+            # drop the file manifest given a subsequent read/write would
+            # need it.
+            # The `LocalFolderFS._hot_files` is kind of a hack around this
+            # (but we endup leaking old files manifests), we should replace
+            # this by a clean ref counting strategy.
+
+            if access.id not in self._hot_files:
+                self.clear_manifest(access)
             return cpy_access
 
         return _recursive_process_copy_map(copy_map)

